.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,50 @@
+name: codeql
+
+concurrency:
+  # Run only for most recent commit in PRs but for all tags and commits on main
+  # Ref: https://docs.github.com/en/actions/using-jobs/using-concurrency
+  group: ${{ github.workflow }}-${{ github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+  schedule:
+    - cron: '42 1 * * 4'
+  workflow_dispatch: {}
+
+permissions:
+  contents: read
+
+jobs:
+  analyze:
+    timeout-minutes: ${{ fromJSON(vars.GHA_DEFAULT_TIMEOUT || 10) }}
+    name: analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
```

Comment:
TIL 
<img width="757" alt="image" src="https://github.com/user-attachments/assets/f9977a0b-1d36-4406-a999-ea05b5d20ca6" />
[src](https://docs.github.com/en/code-security/code-scanning/introduction-to-code-scanning/about-code-scanning-with-codeql)
it's in public preview for worfklows, but maybe it's still worth to consider 🤔

Reply:
Created an issue in https://github.com/Kong/team-k8s/issues/338 for this. 

Interestingly, neither https://docs.github.com/en/code-security/code-scanning/introduction-to-code-scanning/about-code-scanning-with-codeql nor https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/ mention which keywords to use in `language` field to enable this.